evidently/analyzers/data_drift_analyzer.py

```diff
@@ -22,28 +24,94 @@ def dataset_drift_evaluation(p_values, drift_share=0.5):
 PValueWithConfidence = collections.namedtuple("PValueWithConfidence", ["p_value", "confidence"])
 
 
+@dataclass
+class DataDriftAnalyzerFeatureMetrics:
+    current_small_hist: list
+    ref_small_hist: list
+    feature_type: str
+    p_value: float
+
+    def as_dict(self) -> Dict[str, Any]:
+        return {
+            'current_small_hist': self.current_small_hist,
+            'ref_small_hist': self.ref_small_hist,
+            'feature_type': self.feature_type,
+            'p_value': self.p_value,
+        }
+
+
+@dataclass
+class DataDriftAnalyzerMetrics:
+    n_features: int
+    n_drifted_features: int
+    share_drifted_features: float
+    dataset_drift: bool
+    features: Dict[str, DataDriftAnalyzerFeatureMetrics]
+
+    def as_dict(self) -> Dict[str, Any]:
+        result: Dict[str, Any] = {
+            'n_features': self.n_features,
+            'n_drifted_features': self.n_drifted_features,
+            'share_drifted_features': self.share_drifted_features,
+            'dataset_drift': self.dataset_drift
+        }
+
+        for feature_name, feature_metrics in self.features.items():
+            result[feature_name] = feature_metrics.as_dict()
+
+        return result
+
+
+@dataclass
+class DataDriftAnalyzerResults:
+    utility_columns: DatasetUtilityColumns
+    cat_feature_names: List[str]
+    num_feature_names: List[str]
+    target_names: Optional[List[str]]
+    options: DataDriftOptions
+    metrics: DataDriftAnalyzerMetrics
+
+    def as_dict(self) -> Dict[str, Any]:
+        """Convert the analyser data to dict for data serialization"""
+        return {
+            'utility_columns': self.utility_columns.as_dict(),
+            'cat_feature_names': self.cat_feature_names,
+            'num_feature_names': self.num_feature_names,
+            'target_names': self.target_names,
+            'options': self.options.as_dict(),
+            'metrics': self.metrics.as_dict()
+        }
+
+    def get_all_features_list(self) -> List[str]:
+        """List all features names"""
+        return self.cat_feature_names + self.num_feature_names
+
+
 class DataDriftAnalyzer(Analyzer):
-    def calculate(self, reference_data: pd.DataFrame, current_data: pd.DataFrame, column_mapping: ColumnMapping):
+    @staticmethod
+    def get_data_drift_results(analyzer_results) -> DataDriftAnalyzerResults:
```

Comment:
I think here would be better to rename it so it won't repeat itself.
```suggestion
    def get_results(analyzer_results) -> DataDriftAnalyzerResults:
```
Is there a way to place such method to base Analyzer? So it wont be necessary to write it individually for each Analyzer.

Reply:
> I think here would be better to rename it so it won't repeat itself.
Yes, my bad,  just some overthinking, fixed

Reply:
About moving the method to base Analyser - need to investigate ways for types calculations with other analyzators